theano/tensor/basic.py

```diff
@@ -4871,11 +4876,11 @@ def c_code(self, node, name, inputs, outputs, sub):
             fail = sub['fail']
             return """
             assert (PyArray_NDIM(%(shp)s) == 1);
-            npy_intp new_dims[%(new_ndim)s];
+            npy_intp new_dims[%(params)s->ndim];
```

Comment:
The string here is intended to be formatted, using the Python old-style formatting: every occurence of `%(varname)s` will be replace with value of a field `varname` in a dict passed as formatting argument (`string_to_format % dict_with_fields`). Here the dict passed for formatting is `locals()` that contains any local variable defined into current function.

So, as you add `%(params)s` in the string, there should be a variable named `params` in the code of current function, but you have not added that variable. An error should occur if you test your code.

`params` must contain the name of the C object representing the op param. It is located into `sub`:
```python
params = sub['params']
```